include/pkcs11.h

```diff
@@ -241,6 +241,20 @@ extern "C" {
  * function prototypes. */
 #include "pkcs11f.h"
 
+/* CK_NSS_C_INITIALIZE_ARGS provides the optional arguments to
+ * the modified C_Initialize used by NSS */
+ typedef struct CK_NSS_C_INITIALIZE_ARGS {
+  CK_CREATEMUTEX CreateMutex;
+  CK_DESTROYMUTEX DestroyMutex;
+  CK_LOCKMUTEX LockMutex;
+  CK_UNLOCKMUTEX UnlockMutex;
+  CK_FLAGS flags;
+  CK_VOID_PTR LibraryParameters;
+  CK_VOID_PTR pReserved;
+} CK_NSS_C_INITIALIZE_ARGS;
+
+typedef CK_NSS_C_INITIALIZE_ARGS CK_PTR CK_NSS_C_INITIALIZE_ARGS_PTR;
+

```

Comment:
Since `pkcs11.h` is vendored directly from the standard, I think it would make sense to put these parameters in an NSS-specific header file.  What do you think?

Reply:
That sounds sensible